### PR TITLE
Use docker/distribution and recreate Gopkg.toml

### PR DESCRIPTION
**Pull request by metalmatze**
I can now compile this again.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -141,110 +76,47 @@
...
   revision = "e9f42fa127660e552d0ad2b589868d403a9be7c6"
   version = "v1.1.1"
 
 [[projects]]
-  branch = "master"
   name = "github.com/kr/pretty"
   packages = ["."]
-  revision = "cfb55aafdaf3ec08f0db22699ab822c50091b1c4"
+  revision = "73f6ac0b30a98e433b289500d779f50c1a6f0712"
+  version = "v0.1.0"
```
Comment (tboerger) [inline Gopkg.lock]: Downgrade to specific version might break something

File: Gopkg.lock [hunk 2]
```diff
@@ -141,110 +76,47 @@
...
 
 [[projects]]
-  branch = "master"
   name = "github.com/kr/text"
   packages = ["."]
-  revision = "7cafcd837844e784b526369c9bce262804aebc60"
+  revision = "e2ffdb16a802fe2bb95e2e35ff34f0e53aeef34f"
+  version = "v0.1.0"
```
Comment (tboerger) [inline Gopkg.lock]: Downgrade to specific version might break something

File: Gopkg.lock (hunk 1 above)
Comment (metalmatze) [reply Gopkg.lock]: True. I just checked this one. 0.1.0 is 15 days old. Should be fine.

File: Gopkg.lock (hunk 2 above)
Comment (metalmatze) [reply Gopkg.lock]: True. I just checked this one. 0.1.0 is 15 days old. Should be fine.
The hash is 2 years old.

File: config/compiler/image.go [hunk 3]
```diff
@@ -19,7 +37,16 @@ func expandImage(name string) string {
 	if err != nil {
 		return name
 	}
-	return reference.WithDefaultTag(ref).String()
+	s := ref.String()
+
+	if !strings.Contains(s, ":") {
+		s = fmt.Sprintf("%s:%s", s, defaultTag)
```
Comment (bradrydzewski) [inline config/compiler/image.go]: is this passing unit tests?  this function should return an image name with no tag, and this block looks like it adds a default tag if one doesn't exit.

Comment (bradrydzewski) [reply config/compiler/image.go]: it looks like there is a `TrimNamed` function in the docker distribution package that we could potentially use here. https://godoc.org/github.com/docker/distribution/reference#TrimNamed

File: config/compiler/image.go [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+	legacyDefaultHostname = "index.docker.io"
+	// defaultRepoPrefix is the prefix used for default repositories in default host
+	defaultRepoPrefix = "library"
 )
 
 // trimImage returns the short image name without tag.
 func trimImage(name string) string {
+	name = strings.TrimPrefix(name, legacyDefaultHostname+"/")
+	name = strings.TrimPrefix(name, defaultHostname+"/")
+	name = strings.TrimPrefix(name, defaultRepoPrefix+"/")
```
Comment (bradrydzewski) [inline config/compiler/image.go]: it looks like there is a `reference.Path` function that can be used to trim the path from the image https://godoc.org/github.com/docker/distribution/reference#Path

**Comment by bradrydzewski** [comment]
turns out I hadn't hooked up CI yet, but it looks like the unit tests are failing

```
--- FAIL: Test_trimImage (0.00s)
	image_test.go:57: Want image "golang:latest" trimmed to "golang", got "golang:latest"
	image_test.go:57: Want image "golang:1.0.0" trimmed to "golang", got "golang:1.0.0"
	image_test.go:57: Want image "library/golang" trimmed to "golang", got "library/golang"
	image_test.go:57: Want image "library/golang:latest" trimmed to "golang", got "library/golang:latest"
	image_test.go:57: Want image "library/golang:1.0.0" trimmed to "golang", got "library/golang:1.0.0"
	image_test.go:57: Want image "index.docker.io/library/golang:1.0.0" trimmed to "golang", got "index.docker.io/library/golang:1.0.0"
	image_test.go:57: Want image "docker.io/library/golang:1.0.0" trimmed to "golang", got "docker.io/library/golang"
--- FAIL: Test_expandImage (0.00s)
	image_test.go:112: Want image "golang" expanded to "golang:latest", got "golang"
	image_test.go:112: Want image "library/golang" expanded to "golang:latest", got "library/golang"
	image_test.go:112: Want image "library/golang:latest" expanded to "golang:latest", got "library/golang:latest"
	image_test.go:112: Want image "library/golang:1.0.0" expanded to "golang:1.0.0", got "library/golang:1.0.0"
	image_test.go:112: Want image "index.docker.io/library/golang:1.0.0" expanded to "golang:1.0.0", got "index.docker.io/library/golang:1.0.0"
```

**Comment by bradrydzewski** [comment]
I'm realizing this migration to `docker/distribution` might not be so simple. The new docker library fails to parse some valid image names that the old library was able to parse.

This function:

```
ref, err := reference.Parse(image)
```

Fails to parse thes inputs:

```
"golang"
"golang:latest"
"library/golang:latest"
```

and returns an error

**Comment by bradrydzewski** [comment]
I fixed it locally ... thanks for doing the initial leg work :)

**Comment by metalmatze** [comment]
Strange, I let the unit tests run locally and they all passed. :thinking: :wink:

**Pull request merged**